x-pack/plugins/fleet/server/services/epm/data_streams/get.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+
+import type { PackageDataStreamTypes } from '../../../../common/types';
+
+import { dataStreamService } from '../../data_streams';
+
+export async function getDataStreams(options: {
+  esClient: ElasticsearchClient;
+  type?: PackageDataStreamTypes;
+  datasetQuery?: string;
+  sortDirection: 'asc' | 'desc';
+  uncategorisedOnly: boolean;
+}) {
+  const { esClient, type, datasetQuery, uncategorisedOnly, sortDirection } = options;
+
+  const allDataStreams = await dataStreamService.getMatchingDataStreams(esClient, {
+    type: type ? type : '*',
+    dataset: datasetQuery ? `*${datasetQuery}*` : '*',
+  });
+
+  const filteredDataStreams = uncategorisedOnly
```

Comment:
Not sure if this is the best name? But we need some sort of option that can differentiate between wanting all data streams or only wanting uncategorised ones, this will be important for the work within Discover.